dev/WindowsAppRuntime_BootstrapDLL/MddBootstrap.cpp

```diff
@@ -251,7 +282,105 @@ void RemoveFrameworkFromPath(PCWSTR frameworkPath)
+        return true;
+    }
+
+    // Enable the Enumeration-style LifetimeManager if the environment variable
+    // MICROSOFT_WINDOWSAPPRUNTIME_DDLM_ENUMERATION is defined (for testing scenarios)
+    if (GetEnvironmentVariableW(L"MICROSOFT_WINDOWSAPPRUNTIME_DDLM_ENUMERATION", nullptr, 0) > 0)
+    {
+        return true;
+    }
+
+    // Use the AppExtension-style LifetimeManager
+    return false;
+}
+
+void CreateLifetimeManager(
+    UINT32 majorMinorVersion,
+    PCWSTR versionTag,
+    PACKAGE_VERSION minVersion,
+    wil::com_ptr_nothrow<IDynamicDependencyLifetimeManager>& lifetimeManager,
+    wil::unique_event& endTheLifetimeManagerEvent,
+    wil::unique_cotaskmem_string& ddlmPackageFullName)
+{
+    if (IsLifetimeManagerViaEnumeration())
+    {
+        CreateLifetimeManagerViaEnumeration(majorMinorVersion, versionTag, minVersion, endTheLifetimeManagerEvent, ddlmPackageFullName);
+    }
+    else
+    {
+        CreateLifetimeManagerViaAppExtension(majorMinorVersion, versionTag, minVersion, lifetimeManager, ddlmPackageFullName);
+    }
+}
+
+void CreateLifetimeManagerViaAppExtension(
+    UINT32 majorMinorVersion,
+    PCWSTR versionTag,
+    PACKAGE_VERSION minVersion,
+    wil::com_ptr_nothrow<IDynamicDependencyLifetimeManager>& lifetimeManager,
+    wil::unique_cotaskmem_string& ddlmPackageFullName)
+{
+    const auto appDynamicDependencyLifetimeManagerClsid{ FindDDLMViaAppExtension(majorMinorVersion, versionTag, minVersion) };
+
+    wil::com_ptr_nothrow<IDynamicDependencyLifetimeManager> dynamicDependencyLifetimeManager{
+        wil::CoCreateInstance<IDynamicDependencyLifetimeManager>(appDynamicDependencyLifetimeManagerClsid, CLSCTX_LOCAL_SERVER)
+    };
+
+    THROW_IF_FAILED(dynamicDependencyLifetimeManager->Initialize());
+
+    wil::unique_cotaskmem_string packageFullName;
+    THROW_IF_FAILED(dynamicDependencyLifetimeManager->GetPackageFullName(&packageFullName));
+
+    lifetimeManager = std::move(dynamicDependencyLifetimeManager);
+    ddlmPackageFullName = std::move(packageFullName);
+}
+
+void CreateLifetimeManagerViaEnumeration(
+    UINT32 majorMinorVersion,
+    PCWSTR versionTag,
+    PACKAGE_VERSION minVersion,
+    wil::unique_event& endTheLifetimeManagerEvent,
+    wil::unique_cotaskmem_string& ddlmPackageFullName)
+{
+    std::wstring packageFamilyName;
+    std::wstring packageFullName;
+    FindDDLMViaEnumeration(majorMinorVersion, versionTag, minVersion, packageFamilyName, packageFullName);
+
+    // Create the named event used later to signal to the lifetime manager it's time to quit
+    // The named event has the syntax: "<processid>;<packagefullname>;<uniqueid>"
+    GUID uniqueId{};
+    THROW_IF_FAILED(CoCreateGuid(&uniqueId));
+    const auto c_uniqueIdAsString{ winrt::to_hstring(uniqueId) };
+    auto eventName{ wil::str_printf<wil::unique_cotaskmem_string>(L"%u;%s;%s", GetCurrentProcessId(), packageFullName.c_str(), c_uniqueIdAsString.c_str()) };
```

Comment:
does this need to have "Global\" prefix for inter-session access?

Reply:
Do OOP COM servers offer inter-session access?

This is per-user and it's almost impossible to have the same user logged in twice, or at least it used to be. Has that changed lately?